src/parser.ts

```diff
@@ -2583,12 +2583,26 @@ export class Parser {
     parseThrowStatement(): Node.ThrowStatement {
         const node = this.createNode();
         this.expectKeyword('throw');
+        const hasLineTerminator = this.hasLineTerminator;
 
-        if (this.hasLineTerminator) {
-            this.throwError(Messages.NewlineAfterThrow);
+        let argument;
+        try {
+            argument = this.parseExpression();
+        } catch (e) {
+            if (!hasLineTerminator) {
+                throw e;
+            }
+        }
+
+        if (hasLineTerminator) {
+            const shouldThrow = !argument ||
+                (argument && argument.type !== 'TemplateLiteral');
+
+            if (shouldThrow) {
+                this.throwError(Messages.NewlineAfterThrow);
+            }
         }
```

Comment:
I don't think this logic handles the following case (expected to still thrown an exception):
```
throw
`foo
bar`
```